core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -318,7 +318,7 @@ void exitIdleMode() {
 
   // Must be run from channelExecutor
   private void cancelIdleTimer() {
-    if (idleModeTimerFuture != null) {
+    if (idleModeTimerFuture != null && idleModeTimer != null) {
```

Comment:
Damn! I didn't intend to include this, I was experimenting with some tooling and this slipped somehow. Sorry, reverted.